components/Tabs/tests/unit/Theming/TabsThemingTests.m

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import <MaterialComponents/MaterialColorScheme.h>
+#import <MaterialComponents/MaterialTabs.h>
+#import <MaterialComponents/MaterialTypographyScheme.h>
+#import "MaterialContainerScheme.h"
+#import "MaterialTabs+Theming.h"
+
+static const CGFloat kUnselectedTitleOpacity = (CGFloat)0.6;
+static const CGFloat kUnselectedImageOpacity = (CGFloat)0.54;
+static const CGFloat kBottomDividerOpacity = (CGFloat)0.12;
+
+@interface MDCTabsThemingTest : XCTestCase
+@property(nonatomic, strong) MDCTabBar *tabBar;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
+@property(nonatomic, strong) MDCTypographyScheme *typographyScheme;
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@end
+
+@implementation MDCTabsThemingTest
+
+- (void)setUp {
+  [super setUp];
+
+  self.tabBar = [[MDCTabBar alloc] init];
+  self.colorScheme =
+      [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  self.typographyScheme =
+      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  self.containerScheme = [[MDCContainerScheme alloc] init];
+  self.containerScheme.colorScheme = self.colorScheme;
+  self.containerScheme.typographyScheme = self.typographyScheme;
+}
+
+- (void)tearDown {
+  self.tabBar = nil;
+  self.colorScheme = nil;
+  self.typographyScheme = nil;
+  self.containerScheme = nil;
+
+  [super tearDown];
+}
+
+- (void)testTabBarPrimaryThemingDefault {
+  // When
+  [self.tabBar applyPrimaryThemeWithScheme:self.containerScheme];
+
+  // Then
+  [self verifyTabBarPrimaryTheming];
+}
+
+- (void)testTabBarPrimaryThemingCustom {
+  // When
+  self.colorScheme.primaryColor = [UIColor colorWithWhite:(CGFloat)0.9 alpha:0];
```

Comment:
Tip: Please make the custom schemes a method or property you can assign to the container scheme.  They should also be part of the `// Given` and the `// When` is only applying the theming extension.  This will make the tests much easier to read and reduce repeated code.

```objc
// Given
self.containerScheme.colorScheme = [self customColorScheme];
self.containerScheme.typographyScheme = [self customTypographyScheme];

// When
[self.tabBar applyPrimaryThemeWithScheme self.containerScheme];
```